src/main/java/com/iridium/iridiumskyblock/managers/SchematicManager.java

```diff
@@ -115,4 +125,78 @@ private CompletableFuture<Void> pasteSchematic(Island island, Schematics.Schemat
+            IridiumSkyblock.getInstance().getSchematics().failSound.play(player);
+            return false;
+        }
+
+        purchase(player, schematic);
+
+        IridiumSkyblock.getInstance().getSchematics().successSound.play(player);
+
+        List<Placeholder> bankPlaceholders = IridiumSkyblock.getInstance().getBankItemList().stream()
+                .map(BankItem::getName)
+                .map(name -> new Placeholder(name + "_cost", formatPrice(getBankBalance(player, name))))
+                .collect(Collectors.toList());
+
+        player.sendMessage(StringUtils.color(StringUtils.processMultiplePlaceholders(IridiumSkyblock.getInstance().getMessages().paidForRegen
+                        .replace("%prefix%", IridiumSkyblock.getInstance().getConfiguration().prefix)
+                        .replace("%player%", player.getName())
+                        .replace("%schematic%", StringUtils.color(schematic.item.displayName))
+                        .replace("%vault_cost%", formatPrice(schematic.regenCost.money)),
+                bankPlaceholders)
+        ));
+
+        return true;
+    }
+
+    private double getBankBalance(Player player, String bankItem) {
+        User user = IridiumSkyblock.getInstance().getUserManager().getUser(player);
+        return IridiumSkyblock.getInstance().getTeamManager().getTeamViaID(user.getTeamID())
+                .map(team -> IridiumSkyblock.getInstance().getTeamManager().getTeamBank(team, bankItem))
+                .map(TeamBank::getNumber)
+                .orElse(0.0);
+    }
+
+    private void setBankBalance(Player player, String bankItem, double amount) {
+        User user = IridiumSkyblock.getInstance().getUserManager().getUser(player);
+        Optional<Island> island = IridiumSkyblock.getInstance().getTeamManager().getTeamViaID(user.getTeamID());
+        if (!island.isPresent()) return;
+        IridiumSkyblock.getInstance().getTeamManager().getTeamBank(island.get(), bankItem).setNumber(amount);
+    }
+
+    private boolean canPurchase(Player player, Schematics.SchematicConfig schematic) {
+        double moneyCost = schematic.regenCost.money;
+        Economy economy = IridiumSkyblock.getInstance().getEconomy();
+        for (String bankItem : schematic.regenCost.bankItems.keySet()) {
+            double cost = round(schematic.regenCost.bankItems.get(bankItem), 2);
+            if (getBankBalance(player, bankItem) < cost) return false;
+        }
+
+        return moneyCost == 0 || economy != null && economy.getBalance(player) >= schematic.regenCost.money;
+    }
+
+    private void purchase(Player player, Schematics.SchematicConfig schematic) {
+        IridiumSkyblock.getInstance().getEconomy().withdrawPlayer(player, schematic.regenCost.money);
+
+        for (String bankItem : schematic.regenCost.bankItems.keySet()) {
+            setBankBalance(player, bankItem, getBankBalance(player, bankItem) - round(schematic.regenCost.bankItems.get(bankItem), 2));
```

Comment:
not a fan of this since you retrieve the object twice, its probs not a huge difference, but technically you are wasting performance by doing the same operation twice to recieve the same object

Reply:
i think im stupid, what object

Reply:
They both call the [IslandManager#getTeamBank](https://github.com/Iridium-Development/IridiumSkyblock/blob/6292d20aa2111933b7970d2e739db43befb45af4/src/main/java/com/iridium/iridiumskyblock/managers/IslandManager.java#L330) method, whilst this shouldnt be too bad performance wise, depending on how many islands you have, it may have to search through a few thousand entries, so we should try to minimise this where possible

Reply:
ok this might be more of a coding question then; if we set bank as a variable, which is how i would go about solving this, does the code run that once to determine the value and keep that until we update it ourselves, or does calling the variable call the method every time we call it?